devicemodel/hw/mmio/core.c

```diff
@@ -126,10 +128,16 @@ int init_mmio_dev(struct vmctx *ctx, struct mmio_dev_ops *ops, struct acrn_mmiod
 	int ret;
 	uint32_t base;
 
-	ret = mmio_dev_alloc_gpa_resource32(&base, mmiodev->size);
-	if (ret < 0)
-		return ret;
-	mmiodev->base_gpa = base;
+	if (mmiodev->base_gpa == 0UL) {
+	/* TODO: mmio_dev_alloc_gpa_resource32 is only for pure MMIO devie and we should make
+	 * sure the resources of ACPI devices are not in the MMIO resource.
+	 */
```

Comment:
FIXME: The mmio_dev_alloc_gpa_resource32 needs to add one new parameter to indicate if the caller needs one specific GPA instead of dynamic allocation.

Reply:
Please update the FIXME to replace your TODO.

Reply:
Add FIXME